File: java/micronaut/mongodb-sample/src/main/java/mongodb/sample/data/Stock.java
package mongodb.sample.data;

public class Stock {
    private String id;
    private int quantity;

    public Stock() {}

    public Stock(String id, int quantity) {
        this.id = id;
        this.quantity = quantity;
    }

    public String getId() {
        return id;
    }

    public int getQuantity() {
        return quantity;
    }
}

File: scala/DDD_functional-reactive/stock_sample1/src/main/scala/sample/model/Stock.scala
package sample.model

sealed trait Stock {
  val stockId: StockId
  val itemId: ItemId
  val in: Quantity
  val out: Quantity
}

sealed case class UnmanagedStock(stockId: StockId, itemId: ItemId,
                                 in: Quantity = 0, out: Quantity = 0) extends Stock

sealed case class ManagedStock (stockId: StockId, itemId: ItemId,
                                in: Quantity = 0, out: Quantity = 0) extends Stock {
  val qty: Quantity = in - out
}
